shortfin/python/shortfin_apps/llm/components/batcher.py

```diff
@@ -168,13 +174,53 @@ def board(
             if request is not None:
                 exec_requests.append(request)
 
-        exec_process = self.make_invoker(page_cache, fiber, exec_requests)
+        group_id = str(uuid4())
+        self.active_requests[group_id] = exec_requests
+        exec_process = self.make_invoker(page_cache, fiber, exec_requests, group_id)
 
         # We've filled our flight. Remove from the boarding area.
         if exec_requests:
             # And takeoff.
             exec_process.launch()
 
+    def completion_callback(
```

Comment:
Callback interfaces should be defined within the invoker then overridden / handled. E.g. the responder classes the cli/fastapi server is an example of this. Avoid defining out in a class and forcing the owned class to adopt.